models/user.rb

```diff
@@ -61,12 +61,17 @@ def to_hash(params={})
           ).count
 
           # Note that the comments may have been responses to a thread not started by author_id.
-          comment_thread_ids = Comment.where(
+
+          # We need to include the comment_thread in thie query to prevent an N+1 query
+          # when we iterate through all of the comments to get their thread context
+          comment_thread_ids = Comment.includes(:comment_thread).where(
```

Comment:
context only exists on the `comment_thread` so in order to filter out `comments` by the context of their parent `comment_thread` we need to load them as well to prevent N+1 issues when we iterate through the list of comments